aea/protocols/dialogue/base.py

```diff
@@ -857,30 +858,74 @@ def _basic_validation_non_initial_message(
                 ),
             )
 
-        last_message_id = self.last_message.message_id  # type: ignore
-        if message_id != last_message_id + 1:
-            return (
-                False,
-                "Invalid message_id. Expected {}. Found {}.".format(
-                    last_message_id + 1, message_id
-                ),
-            )
         if target < 1:
             return (
                 False,
                 "Invalid target. Expected a value greater than or equal to 1. Found {}.".format(
                     target
                 ),
             )
-        if last_message_id < target:
+
+        if is_outgoing:
+            if message_id != self.get_next_message_id():
+                return (
+                    False,
+                    "Invalid message_id. Expected  message id {} > {}.".format(
+                        message_id, self.get_next_message_id()
+                    ),
+                )
+
+            if not self.last_incoming_message:
+                return False, "Can not reply cause not incoming messages"
+
+            if target > self.last_incoming_message.message_id:
```

Comment:
Insufficient. Validity of targets can only be checked in conjunction with the dialogue reply structure (see GDoc).